ebmlite/core.py

```diff
@@ -811,11 +813,22 @@ def __repr__(self):
         return "<%s %r at 0x%08X>" % (self.__class__.__name__, self.name,
                                       id(self))
 
+    def __enter__(self):
+        """ Enter context manager for this document.
+        """
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """ Close this document on exiting context manager.
+        """
+        self.close()
+
     def close(self):
-        """ Close the EBML file. Should generally be used only if the object
-            was created using a filename, rather than a stream.
+        """ Close the EBML file. If the object doesn't own its data stream,
+            do nothing.
```

Comment:
I think this docstring could be clearer. If nothing else, I'd not say it 'does nothing.' We will probably want to have it do some cleanup in the future, especially if it doesn't own the stream.

Maybe something more like:
```
""" Closes the EBML file. If the `Document` was created using a file/stream (as
    opposed to a filename or raw data), the source file/stream is not closed. 
"""
```

Reply:
so it's kind of weird because what you said is *generally* true but not *strictly* true. Specifically from the context of *just* the `Document` object, the only thing that determines how the stream gets cleaned up is the `ownsStream` parameter that gets passed in at construction. The filename-vs-fileobj determination that you described happens in `Schema.load`, but constructing `Document` outside of that function (which *should* never happen but still) breaks that guarantee.

Should the file ownership bit from `Schema.load` be moved into `Document.__init__`? if we do so then what you say will be *strictly* true (assuming nobody maliciously mutates "private" members), and tbh just feels right anyway.

Reply:
I think it's better to be clear at the expense of some technical minutia. I see the `ownsStream` mostly as internal plumbing. Users explicitly instantiating `Document` objects isn't prohibited, but it's difficult and I see it as an unusual edge case. If you feel strongly about it, maybe just change the second sentence a little:
```python
""" Closes the EBML file. If the `Document` was created via `Schema.load()` using a file/stream ...
"""
```

Reply:
is 73357d943a0dc9f71e631c4a23e9a838326e08a6 better?